sprintvotingapp/sprint/views.py

```diff
@@ -0,0 +1,536 @@
+            )
+
+    @verify_token
+    def put(self, request, id):
+        """
+        For Updating the Sprint Cycle
+        :param request:
+        :return:
+        """
+        try:
+            user = User.objects.get(id=request.data.get("user_id"))
+            if user.is_superuser:
+                sprint = Sprint.objects.get(id=id)
+                serializer = SprintSerializer(sprint, data=request.data)
+                serializer.is_valid(raise_exception=True)
+                serializer.save()
+
+                return Response(
+                    {
+                        "message": "Sprint updated successfully",
+                        "data": serializer.data
+                    }, status=status.HTTP_200_OK
+                )
+            return Response(
+                {
+                    "message": "Only admin or superuser can update a sprint"
+
+                }, status=status.HTTP_401_UNAUTHORIZED
+            )
+        except ValidationError:
+            logging.error("Validation failed")
+            return Response(
+                {
+                    "message": "Sprint not updated"
+                },
+                status=status.HTTP_400_BAD_REQUEST)
+
+        except Exception as e:
+            return Response(
+                {
+                    "message": "Error Occurred",
+                    "error": str(e)
+                }, status=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            )
+
+    @verify_token
+    def get(self, request):
+        """
+             this method is created for retrieve data
+             :param request: format of the request
+             :return: Response
+        """
+        try:
+
+            sprint = Sprint.objects.filter(is_active=True)
+            serializer = SprintSerializer(sprint, many=True)
+            if sprint:
+                return Response(
+                    {
+                        "message": "Your active sprints are:",
+                        "data": serializer.data
+                    }, status=status.HTTP_200_OK
+                )
+            return Response(
+                {
+                    "message": "No active sprint",
+                }, status=status.HTTP_404_NOT_FOUND
+            )
+
+        except Exception as e:
+            return Response(
+                {
+                    "message": "Error Occurred",
+                    "error": str(e)
+                }, status=status.HTTP_500_INTERNAL_SERVER_ERROR
+            )
+
+    @verify_token
+    def delete(self, request, id):
+        """
+        This method is created for delete the existing data
+        :param request: format of the request
+        :param id: sprint id
+        :return: Response
+        """
+        try:
+            user = User.objects.get(id=request.data.get("user_id"))
+            if user.is_superuser:
+                sprint = Sprint.objects.get(id=id)
+                sprint.delete()
```

Comment:
similar to above comment, should check if sprint exists before deleting.

Reply:
Done